plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobContainer.java

```diff
@@ -203,12 +205,15 @@ public void deleteBlobIgnoringIfNotExists(String blobName) throws IOException {
                     final ObjectListing finalPrevListing = prevListing;
                     list = SocketAccess.doPrivileged(() -> clientReference.client().listNextBatchOfObjects(finalPrevListing));
                 } else {
+                    final ListObjectsRequest listObjectsRequest = new ListObjectsRequest();
```

Comment:
Had to adjust this as well, we were doing deep listing for S3 incorrectly.

Reply:
Is there a test for this? What's the effect of this bug?

Reply:
Yes this was caught by https://github.com/elastic/elasticsearch/pull/42653/files/574896df379b674b484fc59a75a4c55a33e7ce0b#diff-b71311294a29193871313212fcda17d5R156 

without this change if you had a blob:

`/bla/foo`

and you then listed from the root `/` with prefix `/fo` the result would contain the nested `/bla/foo`.

Reply:
Hmm now I ran into the same issue on CI even with this fix, investigating this now.

Reply:
Fixed in https://github.com/elastic/elasticsearch/pull/42653/commits/29ac16bc7a5d88f0f754c822c892f09af53110b7  (forgot to strip slashes in one spot and it caused the recursive delete in tests to fail ... but only on Minio not real S3 because of https://github.com/elastic/elasticsearch/issues/41814 ... fix incoming for that)

@ywelsch this should be good for another review, thanks!